### ISSUE TITLE
CSV.File seems to not handle pool kwarg correctly

### ISSUE THREAD

**Issue by bkamins** [JuliaData/CSV.jl#829]
Take a 10^7 row table with 100 unique values each having 100_000 cardinality, however:

```
julia> CSV.write("test.csv", DataFrame(x=rand("x" .* string.(1:100), 10^7)));

julia> mapcols(typeof, CSV.read("test.csv", DataFrame))
1×1 DataFrame
 Row │ x
     │ DataType
─────┼────────────────
   1 │ Vector{String}

julia> mapcols(typeof, CSV.read("test.csv", DataFrame, pool=0.58))
1×1 DataFrame
 Row │ x
     │ DataType
─────┼────────────────
   1 │ Vector{String}

julia> mapcols(typeof, CSV.read("test.csv", DataFrame, pool=0.59))
1×1 DataFrame
 Row │ x
     │ DataType
─────┼───────────────────────────────────
   1 │ PooledVector{String, UInt32, Vec…
```
which is clearly unexpected (for `pool` around 0.59 the behavior switches).

This kills H2O benchmarks.

CC @quinnj, @nalimilan

**Comment by quinnj** [comment]
Yes, the current algorithm looks at the first 100 rows (POOL_SAMPLE_SIZE) and takes that as a "sample" to determine whether the column should be pooled or not. I have plans to improve this where we first do a sample across all thread chunks for column type, pooled or not, length of strings, etc.

**Comment by bkamins** [comment]
OK - so for now we have update the reader codes in H2O benchmarks.

**Comment by quinnj** [comment]
Will be resolved in #837
